pycalphad/core/minimizer.pyx

```diff
@@ -281,48 +300,49 @@ cdef void fill_equilibrium_system(double[::1,:] equilibrium_matrix, double[::1]
     equilibrium_rhs[system_amount_index] -= system_residual
 
 
 cdef class SystemSpecification:
     def __init__(self, int num_statevars, int num_components, double prescribed_system_amount,
-                   double[::1] initial_chemical_potentials, double[::1] prescribed_elemental_amounts,
-                   int[::1] prescribed_element_indices, int[::1] free_chemical_potential_indices,
+                   double[::1] initial_chemical_potentials, double[:, ::1] prescribed_mole_fraction_coefficients,
+                   double[::1] prescribed_mole_fraction_rhs, int[::1] free_chemical_potential_indices,
                    int[::1] free_statevar_indices, int[::1] fixed_chemical_potential_indices,
                    int[::1] fixed_statevar_indices, int[::1] fixed_stable_compset_indices):
         self.num_statevars = num_statevars
         self.num_components = num_components
         self.prescribed_system_amount = prescribed_system_amount
         self.initial_chemical_potentials = initial_chemical_potentials
-        self.prescribed_elemental_amounts = prescribed_elemental_amounts
-        self.prescribed_element_indices = prescribed_element_indices
+        self.prescribed_mole_fraction_coefficients = prescribed_mole_fraction_coefficients
+        self.prescribed_mole_fraction_rhs = prescribed_mole_fraction_rhs
         self.free_chemical_potential_indices = free_chemical_potential_indices
         self.free_statevar_indices = free_statevar_indices
         self.fixed_chemical_potential_indices = fixed_chemical_potential_indices
         self.fixed_statevar_indices = fixed_statevar_indices
         self.fixed_stable_compset_indices = fixed_stable_compset_indices
         self.max_num_free_stable_phases = num_components + len(free_statevar_indices) - len(fixed_stable_compset_indices)
 
-        # Assuming the prescribed_elemental_amounts doesn't change, this is
+        # Assuming the prescribed_mole_fraction_rhs doesn't change, this is
         # constant and we can keep extra computation (especially calls into
         # NumPy out of the run loop)
-        if self.prescribed_elemental_amounts.shape[0] > 0:
-            self.ALLOWED_MASS_RESIDUAL = min(1e-8, np.min(self.prescribed_elemental_amounts)/10.0)
+        if self.prescribed_mole_fraction_rhs.shape[0] > 0:
+            # With linear combinations of conditions, RHS can now be exactly zero
+            # This means the smallest allowed mass residual needs to be limited to prevent instability
+            self.ALLOWED_MASS_RESIDUAL = max(1e-12, min(1e-8, np.min(np.abs(self.prescribed_mole_fraction_rhs))/10.0))
             # Also adjust mass residual if we are near the edge of composition space
-            self.ALLOWED_MASS_RESIDUAL = min(self.ALLOWED_MASS_RESIDUAL, (1-np.sum(self.prescribed_elemental_amounts))/10.0)
+            self.ALLOWED_MASS_RESIDUAL = min(self.ALLOWED_MASS_RESIDUAL, (1-np.sum(np.abs(self.prescribed_mole_fraction_rhs)))/10.0)
```

Comment:
While I'm thinking about it, do we have examples for linear combination of conditions in the docs?

Reply:
Added